Source/WebCore/html/HTMLMetaElement.cpp

```diff
@@ -192,6 +209,10 @@ void HTMLMetaElement::process()
         document().processReferrerPolicy(contentValue, ReferrerPolicySource::MetaTag);
     else if (equalLettersIgnoringASCIICase(nameValue, "confluence-request-time"_s))
         document().quirks().setNeedsToCopyUserSelectNoneQuirk();
+#if ENABLE(DARK_MODE_CSS)
+    else
+        maybeCallMetaElementColorSchemeChanged(document(), nameValue);
```

Comment:
I would suggest updating the order of logic in `::process` to allow for a null `content`/`name` when processing `color-scheme`. This is similar to how the `http_equivAttr` is checked before the `name` null-check.

The current approach is more difficult to follow due to the use of `maybe`. It also increases the size of all `<meta>` elements with a new member, and I think that could be avoided.

Reply:
That's what I had in mind with https://github.com/WebKit/WebKit/pull/17845#discussion_r1341696812, my apologies if I wasn't clear enough.

Reply:
OK, thanks much — I updated the order of logic in `::process`. But note that I also updated the parameter list for `::process` itself, to pass in the old value of the `name` attribute when it changes. Otherwise, without doing that, I can’t see how `::process` could handle the `name`-removal case.

I did also add a `::isNameColorScheme` member function — to avoid the repetition/verbosity of `equalLettersIgnoringASCIICase(nameValue, "color-scheme"_s) || equalLettersIgnoringASCIICase(nameValue, "supported-color-schemes"_s`.